clients/src/main/java/org/apache/kafka/clients/ClientUtils.java

```diff
@@ -108,10 +108,15 @@ public static ChannelBuilder createChannelBuilder(AbstractConfig config, Time ti
 
     static List<InetAddress> resolve(String host, ClientDnsLookup clientDnsLookup) throws UnknownHostException {
         InetAddress[] addresses = InetAddress.getAllByName(host);
-        if (ClientDnsLookup.USE_ALL_DNS_IPS == clientDnsLookup) {
-            return filterPreferredAddresses(addresses);
-        } else {
-            return Collections.singletonList(addresses[0]);
+        switch (clientDnsLookup) {
+            case DEFAULT:
+                log.warn("Configuration 'client.dns.lookup' value 'default' is deprecated and will be removed in the future version.");
+                return Collections.singletonList(addresses[0]);
+            case USE_ALL_DNS_IPS:
+            case RESOLVE_CANONICAL_BOOTSTRAP_SERVERS_ONLY:
+                return filterPreferredAddresses(addresses);
```

Comment:
Can we add a test that passes `RESOLVE_CANONICAL_BOOTSTRAP_SERVERS_ONLY` showing that we now get multiple addresses instead of only the first one. That's one of the changes in this PR.

Reply:
Added.